endpoint/endpoint.go

```diff
@@ -26,6 +26,14 @@ const (
 	OwnerLabelKey = "owner"
 )
 
+// TTL is a structure defining the TTL of a DNS record
+type TTL int64
+
+// IsConfigured returns true if TTL is configured, false otherwise
+func (ttl TTL) IsConfigured() bool {
+	return ttl > 0
```

Comment:
Can we try to avoid this? Seems like this is the reason that we need `shouldUpdateTTL` for instance, which looks quite complicated to me.

What happens if we just use a plain `int64` and solely rely on its value. Which means, in `NewEndpoint` we probably want to default it to `300` instead of `0`. Then we can just use it as is and there shouldn't be a need for `shouldUpdateTTL`. What do you think? Did you try this before?

Reply:
@linki any comments? :point_up:

Reply:
@kevinjqiu This is how I understood it and it's good thinking. I was just wondering if the added code is worth it. It might even make sense to abstract from those values as we want to hide some provider specific details behind the Kubernetes API. What do you think @ideahitme?

Reply:
@ideahitme any opinions? :point_up:

Reply:
I like this approach, and I think it makes setting TTL more explicit, but maybe using some other constant, like `-1` would be better to actually signify that TTL was not set.

Reply:
@linki and me discussed that we let this change like this

Reply:
Keep in mind that this will change existing TTLs for anybody who's going to update ExternalDNS (if the cloud provider's default TTL is not 300). I don't see a big issue with this. Anyone does?

Reply:
Hmm, no it doesn't, right @kevinjqiu?

Reply:
It doesn't. It keeps the current value. See for more details: https://github.com/kubernetes-incubator/external-dns/pull/320/files/cdb17560e78e4df9f6e474d4c2254f5b6a73487c#r143508870